client/src/components/SearchBar/SearchBar.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react'
+import styled from 'styled-components/macro'
+import { useHistory } from 'react-router-dom'
+
+export default function SearchBar() {
+  const { push } = useHistory()
+  return (
+    <SearchBarForm onSubmit={handleSubmit}>
+      <SearchInput
+        name="q"
+        required
+        placeholder="z.B. Modell oder Marke"
+      ></SearchInput>
+    </SearchBarForm>
+  )
+
+  function handleSubmit(event) {
+    const form = event.target
+    const { q } = form.elements
```

Comment:
IMHO I would write "query" for readability, but in context one can understand the abbreviation.